src/astro/sql/operators/sql_dataframe.py

```diff
@@ -125,35 +126,33 @@ def get_snow_hook(self, table: Table) -> SnowflakeHook:
         )
 
     def _get_dataframe(self, table: Table):
-        conn_type = BaseHook.get_connection(table.conn_id).conn_type
+        database = get_database_from_conn_id(table.conn_id)
         self.log.info(f"Getting dataframe for {table}")
-        if conn_type == "postgres":
-            from psycopg2 import sql
-
+        if database in (Database.POSTGRES, Database.POSTGRESQL):
             self.hook = PostgresHook(
                 postgres_conn_id=table.conn_id, schema=table.database
             )
             schema = table.schema or SCHEMA
             query = (
-                sql.SQL("SELECT * FROM {schema}.{input_table}")
+                postgres_sql.SQL("SELECT * FROM {schema}.{input_table}")
                 .format(
-                    schema=sql.Identifier(schema),
-                    input_table=sql.Identifier(table.table_name),
+                    schema=postgres_sql.Identifier(schema),
+                    input_table=postgres_sql.Identifier(table.table_name),
                 )
                 .as_string(self.hook.get_conn())
             )
             df = self.hook.get_pandas_df(query)
-        elif conn_type == "snowflake":
+        elif database == Database.SNOWFLAKE:
             hook = self.get_snow_hook(table)
             df = hook.get_pandas_df(
                 "SELECT * FROM IDENTIFIER(%(input_table)s)",
                 parameters={"input_table": table.table_name},
             )
-        elif conn_type == "sqlite":
+        elif database == Database.SQLITE:
             hook = SqliteHook(sqlite_conn_id=table.conn_id, database=table.database)
             engine = hook.get_sqlalchemy_engine()
             df = pd.read_sql_table(table.table_name, engine)
```

Comment:
@utkarsharma2 I think that is a great improvement. Since it is a refactor on a part that was not changed in this PR, would it be okay if we have a separate ticket for it?

Reply:
Sure, we can do this later.